homeassistant/components/backup/manager.py

```diff
@@ -0,0 +1,178 @@
+        self.hass = hass
+        self.backup_dir = Path(hass.config.path("backups"))
+        self.backing_up = False
+
+    @property
+    def backups(self) -> dict[str, Backup] | None:
+        """Return a dict of backups."""
+        return self._backups
+
+    def load_backups(self) -> None:
+        """Load data of stored backup files."""
+        backups = {}
+
+        def _read_backup_info(backup_path: Path) -> None:
+            with tarfile.open(backup_path, "r:") as backup_file:
+                if data_file := backup_file.extractfile("./backup.json"):
+                    data = json.loads(data_file.read())
+                    backup = Backup(
+                        slug=data["slug"],
+                        name=data["name"],
+                        date=data["date"],
+                        path=backup_path,
+                        size=round(backup_path.stat().st_size / 1_048_576, 2),
+                    )
+                    backups[backup.slug] = backup
+
+        for backup_path in self.backup_dir.glob("*.tar"):
+            _read_backup_info(backup_path)
+
+        LOGGER.debug("Loaded %s backups", len(backups))
+        self._backups = backups
+
+    def get_backup(self, slug: str) -> Backup | None:
+        """Return a backup."""
+        return None if self._backups is None else self._backups.get(slug)
+
+    def remove_backup(self, slug: str) -> None:
+        """Remove a backup."""
+        LOGGER.error(self._backups)
+        if self._backups is None or (backup := self.get_backup(slug)) is None:
+            return
+        backup.path.unlink(missing_ok=True)
+        LOGGER.debug("Removed backup located at %s", backup.path)
+        self._backups.pop(slug)
+
+    async def generate_backup(self) -> Backup:
+        """Generate a backup."""
+        if self.backing_up:
+            raise HomeAssistantError("Backup already in progress")
+
+        if self._backups is None:
+            self._backups = {}
+
+        try:
+            self.backing_up = True
+            backup_name = f"Core {VERSION}"
+            date_str = now().isoformat()
+            slug = _generate_slug(date_str, backup_name)
+
+            backup_data = {
+                "slug": slug,
+                "name": backup_name,
+                "date": date_str,
+                "type": "partial",
+                "folders": ["homeassistant"],
+                "homeassistant": {} if VERSION.dev else {"version": VERSION},
+                "compressed": True,
+            }
+            tar_file = Path(self.backup_dir, f"{slug}.tar")
+
+            if not self.backup_dir.exists():
+                LOGGER.debug("Creating backup directory")
+                self.hass.async_add_executor_job(self.backup_dir.mkdir)
+
+            def _create_backup() -> None:
+                with TemporaryDirectory() as tmp_dir:
+                    json_util.save_json(
+                        Path(tmp_dir, "backup.json").as_posix(),
```

Comment:
You should use streamed backups like Supervisor. This load everything into memory and don't stream like with old magnet-tapes and save memory or out-off memory issues